src/components/ProjectDomain/addProjectDomain.tsx

```diff
@@ -0,0 +1,225 @@
+"use client";
+import { useSelector } from "react-redux";
+import React, { useEffect, useState } from "react";
+import { Plus } from "lucide-react";
+import { Controller, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+import { RootState } from "@/lib/store";
+import { useToast } from "@/components/ui/use-toast";
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+  DialogTitle,
+  DialogDescription,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  Select,
+  SelectTrigger,
+  SelectItem,
+  SelectValue,
+  SelectContent,
+} from "@/components/ui/select";
+import { statusType } from "@/utils/common/enum";
+import { apiHelperService } from "@/services/projectdomain";
+interface DomainData {
+  _id: string;
+  label: string;
+  description: string;
+  createdAt?: string; // Optional fields
+  createdBy?: string;
+  status?: string;
+}
+
+interface Domain {
+  _id: string;
+  label: string;
+  description: string;
+}
+
+interface AddDomainProps {
+  onAddProjectDomain: (newDomain: DomainData) => void; // Prop to pass the new domain
+}
+
+// Zod schema for form validation
+const domainSchema = z.object({
+  label: z.string().nonempty("Please enter a domain name"),
+  description: z.string().nonempty("Please enter a description"),
+  status: z.enum([statusType.active]).default(statusType.active),
+});
+
+const AddProjectDomain: React.FC<AddDomainProps> = ({ onAddProjectDomain }) => {
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [domains, setDomains] = useState<Domain[]>([]); // Use Domain type here
+  const currentUser = useSelector((state: RootState) => state.user);
+  const currentUserId = currentUser.uid;
+  const { toast } = useToast();
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<DomainData>({
+    resolver: zodResolver(domainSchema),
+    defaultValues: {
+      label: "",
+      description: "",
+      status: statusType.active,
+    },
+  });
+
+  // Fetch the list of domains from the backend
+  useEffect(() => {
+    async function fetchDomains() {
```

Comment:
don't need to call api here if parent component already has this data, pass domains as props and use in child component from there

Reply:
removed